### PR TITLE
Added more trans types for F8/B8 input gemm

### PR DESCRIPTION
**Pull request by solaslin**
Added more transpose types for currently existing F8/B8 gemms which support only TN.

* Added types are **NN/NT/TT**
- [x] F8_B_S
- [x] F8_H_S
---
- [x] F8B8_B_S
- [x] F8B8_H_S
---
- [x] B8F8_B_S (All 4 types)
- [x] B8F8_H_S
---
- [x] B8_B_S (All 4 types)
- [x] B8_H_S (All 4 types)
---
- [x] GTESTs

### CODE REVIEW

File: clients/common/cblas_interface.cpp [hunk 1]
```diff
@@ -287,6 +287,9 @@ CREATEFUNCTION(hipblaslt_f8_fnuz, hipblaslt_f8_fnuz, hip_bfloat16, float, hipbla
 CREATEFUNCTION(hipblaslt_bf8_fnuz, hipblaslt_f8_fnuz, hipblasLtHalf, float, hipblaslt_bf8_fnuz)
 CREATEFUNCTION(hipblaslt_f8_fnuz, hipblaslt_bf8_fnuz, hipblasLtHalf, float, hipblaslt_f8_fnuz)
 CREATEFUNCTION(hipblaslt_f8_fnuz, hipblaslt_bf8_fnuz, hip_bfloat16, float, hipblaslt_f8_fnuz)
+CREATEFUNCTION(hipblaslt_bf8_fnuz, hipblaslt_bf8_fnuz, hipblasLtHalf, float, hipblaslt_bf8_fnuz)
+CREATEFUNCTION(hipblaslt_bf8_fnuz, hipblaslt_bf8_fnuz, hip_bfloat16, float, hipblaslt_bf8_fnuz)
+CREATEFUNCTION(hipblaslt_bf8_fnuz, hipblaslt_f8_fnuz, hip_bfloat16, float, hipblaslt_bf8_fnuz)
 CREATEFUNCTION(hipblasLtHalf, hipblasLtHalf, hipblasLtHalf, float, hipblasLtHalf)
```
Comment (solaslin) [inline clients/common/cblas_interface.cpp:293]: added B8_H_S, B8_B_S, B8F8_B_S in clients

File: clients/gtest/matmul_gtest.yaml [hunk 2]
```diff
@@ -590,8 +590,7 @@ Tests:
   M: [128, 129]
   N: [128, 129]
   K: [128, 129]
-  transA: T
-  transB: N
+  transA_transB: *transA_transB_range
   alpha: 1
```
Comment (solaslin) [inline clients/gtest/matmul_gtest.yaml:594]: test all trans types

File: clients/gtest/matmul_gtest.yaml [hunk 3]
```diff
@@ -601,15 +600,14 @@ Tests:
...
   category: pre_checkin
   function:
     matmul: *real_precisions_1b_dst_bf16
   M: [128, 129]
   N: [128, 129]
   K: [128, 129]
-  transA: T
-  transB: N
+  transA_transB: *transA_transB_range
   alpha: 1
```
Comment (solaslin) [inline clients/gtest/matmul_gtest.yaml:611]: fixed typo and enable all trans types

File: clients/common/cblas_interface.cpp (hunk 1 above)
Comment (solaslin) [reply clients/common/cblas_interface.cpp:293]: Same additions were applied in tensile side (TypedId.hpp)

**Review by KKyang — APPROVED** [review]

**Review by hcman2 — APPROVED** [review]

**Comment by solaslin** [comment]
Will merge this once CI tests are passed

**Pull request merged**